src/Microsoft.Health.Fhir.Core.UnitTests/Features/Search/Expressions/Parsers/SearchValueExpressionBuilderTests.cs

```diff
@@ -584,7 +584,7 @@ public void GivenAReferenceUsingResourceTypeAndResourceIdTargetingInternalResour
             const string resourceId = "123";
             string reference = $"{resourceType}/{resourceId}";
 
-            _referenceSearchValueParser.Parse(reference).Returns(new ReferenceSearchValue(ReferenceKind.Internal, null, resourceType, resourceId));
+            _referenceSearchValueParser.Parse(reference).Returns(new ReferenceSearchValue(ReferenceKind.Internal, null, resourceType.ToString(), resourceId));
```

Comment:
nit: these could just be string from beginning. I will fix them as I refactor the tests.